main.cc

```diff
@@ -244,6 +244,18 @@ verify_rlimit(bool developer_mode) {
     if (r == -1) {
         throw std::system_error(errno, std::system_category());
     }
+
+    // First, try to increase the soft limit to the hard limit
+    // Ref: http://0pointer.net/blog/file-descriptor-limits.html
+
+    if (lim.rlim_cur < lim.rlim_max) {
+        lim.rlim_cur = lim.rlim_max;
+        r = setrlimit(RLIMIT_NOFILE, &lim);
+        if (r == -1) {
+            throw std::system_error(errno, std::system_category());
```

Comment:
Isn't it a little harsh? I realize that even unprivileged users are usually allowed to bump the soft limit up to the hard limit value, but perhaps some esoteric selinux configuration or hardened kernels can still be problematic? A warning should also suffice, since a failure here doesn't really prevent Scylla from booting

Reply:
There should be no selinux restriction here, since we're raising to the hard limit, but a warning instead of throw is safer. I'll adjust.

Reply:
I'm not sure that even a warning is needed. It's not like this can ever happen unless you are running on a wierd OS.
But just make sure that if it failed, you should know that rlim_cur did not change.

Reply:
Demoted to warning. I think ignoring an unexpected failure is bad, it can waste time debugging.